### PR TITLE
Use new @plotly/dash-component-plugins to handle location path changes

### PR DESCRIPTION
**Pull request by Marc-Andre-Rivet**
Linked to https://github.com/plotly/dash/pull/1094

- [ ] changelog entry
- [ ] new dcc.Link / dcc.Location test on link click, browser forward, browser backward

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "@babel/plugin-syntax-dynamic-import": "^7.2.0",
     "@babel/preset-env": "^7.4.1",
     "@babel/preset-react": "^7.0.0",
-    "@plotly/dash-component-plugins": "^1.0.2",
+    "@plotly/dash-component-plugins": "file:plotly-dash-component-plugins-1.1.0-rc1.tgz",
```
Comment (Marc-Andre-Rivet) [inline package.json]: Waiting for `1.1.0`

File: src/components/Link.react.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ export default class Link extends Component {
             window.location.pathname = href;
         } else {
             window.history.pushState({}, '', href);
-            window.dispatchEvent(new CustomEvent('onpushstate'));
+            window.dispatchEvent(new CustomEvent('_dashprivate_onpushstate'));
```
Comment (Marc-Andre-Rivet) [inline src/components/Link.react.js]: Make the link <--> location event explicitly private. No one outside of DCC should listen to this.

File: src/components/Location.react.js [hunk 3]
```diff
@@ -81,23 +84,25 @@ export default class Location extends Component {
...
-        window.addEventListener('onpopstate', listener());
-        window.onpopstate = listener('POP');
+        window.onpopstate = this.onLocationChange;
 
-        // non-standard, emitted by Link.react
-        window.addEventListener('onpushstate', listener());
+        window.addEventListener(
+            '_dashprivate_onpushstate',
+            this.onLocationChange
+        );
```
Comment (Marc-Andre-Rivet) [inline src/components/Location.react.js]: Location triggers a `location changed` event on both (1) click on dcc.Link with `refresh=false (default)`, (2) user clicking `back` and `forward` buttons in the browser UI. Prior to this change it was only possible to listen for `(1)` with `"onpushstate"`

Comment (wbrgss) [reply src/components/Location.react.js]: nice :relaxed: 
the location history feature is working well

File: tests/integration/link/test_link_event.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    )
+
+    history_counter = dash_dcc.driver.execute_script(
+        '''
+        return window._test_history_event_counter;
+    '''
+    )
+
+    assert link_counter == 1
+    assert history_counter == 1
```
Comment (Marc-Andre-Rivet) [inline tests/integration/link/test_link_event.py]: Simple test making sure the events are being triggered by the components

**Pull request merged**